bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/LazyValue.java

```diff
@@ -0,0 +1,50 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2023 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.jetbrains.annotations.Nullable;
+
+
+/**
+ * Resolve a value lazily
+ *
+ * @param <T>
+ */
+
+public class LazyValue<T> {
+
+	private Optional<T> value;
+	private Supplier<T> supplier;
+	
+	public LazyValue(Supplier<T> supplier) {
+		this.supplier = supplier;
+	}
+	
+	@Nullable
+	public T get() {
+		if (value == null) {
+			value = Optional.ofNullable(supplier.get());
+		}
+		return value.get();
```

Comment:
If `supplier.get()` returns `null` then this will cause a `NoSuchElementException`.
I think what we want here is `return value.orElse(null)`

Reply:
good point, fixed